crates/runtime/src/http/v1.rs

```diff
@@ -23,12 +23,7 @@ pub(crate) mod datasets {
     ) -> Json<Vec<Dataset>> {
         let datasets: Vec<Dataset> = match filter.source {
             Some(source) => match source {
-                DatasetSource::Sink => app
-                    .datasets
-                    .iter()
-                    .filter(|d| d.source.is_none())
-                    .cloned()
-                    .collect(),
+                DatasetSource::Sink => app.datasets.clone(),
```

Comment:
Why did you change this?

Reply:
I'll fix this in a follow up PR

Reply:
source is no longer its own optional field. All datasets implicitly have a source now (everything before the first `/` or `local`)

Reply:
So no filtering is needed here

Reply:
On thinking about this, I think we should be even more explicit. If we are generally using domain names like `spice.ai` or `databricks.com` as the "source" then we should use `localhost` as the local source. E.g. `localhost/org/app`

Reply:
If we don't filter here - this the query parameter is pointless and doesn't actually do what I need it to :)

Reply:
Ok, will let you fix it up

Reply:
So then this changes to a filter on source - and to get the datasets that are publishable, I filter to the ones with a `localhost` source.